src/agni/client/HeartbeatMonitor.java

```diff
@@ -0,0 +1,39 @@
+package agni.client;
+
+public class HeartbeatMonitor implements Runnable {
+
+    private long TIMEOUT_PERIOD;
+
+    private long lastHeartbeatTime;
+    private boolean timedOut;
+
+    public HeartbeatMonitor(long timeoutPeriod) {
+        lastHeartbeatTime = System.nanoTime();
+        timedOut = false;
+        TIMEOUT_PERIOD = timeoutPeriod;
+    }
+
+    public void receivedHeartbeat() {
+        lastHeartbeatTime = System.nanoTime();
+    }
+
+    public final boolean isTimedOut() {
+        return timedOut;
+    }
+
+    @Override
+    public void run() {
+        while (true) {
```

Comment:
So I read about while(true) vs for(;;), StackOverFlow seems to think they're identical, and translated to the same lower-level code. But since we were talking about it the other day, I wanted to mention this